### ISSUE TITLE
Button inside Toolbar component complains about several props

### ISSUE THREAD

**Issue by manattersa** [marmelab/react-admin#3245]
**What you were expecting:**
Using a Button component inside a Toolbar to customize a SimpleForm behaviour shouldn't complain about props.

**What happened instead:**
React complains with several "React does not recognize the `<xxx>` prop on a DOM element".

**Steps to reproduce:**
I followed instructions about how to customize the Toolbar for a SimpleForm (https://marmelab.com/react-admin/CreateEdit.html#toolbar) but instead of a SaveButton I use a simple Button in it.

**Related code:**
`
const MyToolbar = props => (
    <Toolbar {...props}>
        <Button label="Test"/>
    </Toolbar>
);
`

**Other information:**
If instead I sanitize the props it works without warnings.
```
const MyButton = ({
    basePath,
    className,
    classes,
    saving,
    invalid,
    variant,
    translate,
    handleSubmit,
    handleSubmitWithRedirect,
    submitOnEnter,
    record,
    redirect,
    resource,
    locale,
    showNotification,
    undoable,
    pristine,
    ...rest
}) => <Button {...rest}/>;

const MyToolbar = props => (
    <Toolbar {...props}>
        <MyButton label="Test"/>
    </Toolbar>
);
```
**Environment**

* React-admin version: 2.9.1
* React version: 16.8.6
* Browser: Firefox, Chrome.

**Comment by Kmaschta** [comment]
Instead of changing the `<Button>` component, we have to change the documentation so that it doesn't cause warnings or errors.

PR are welcome to change the doc!

**Comment by fzaninotto** [comment]
Fixed by #4544
